guides-markdown/multisig-cli.md

```diff
@@ -0,0 +1,463 @@
+
+{
+  "hash": "659800f1c5a8780ec2215afebd4e3f285ca999027ee3a789ad5655786fd04ead",
+  "witnessHash": "3c96d96668caf81c1b797863aeda9d6bd843e461da4511db18e4961f4507c8c2",
+  "fee": 4540,
+  "rate": 23645,
+  "mtime": 1572014394,
+  "version": 1,
+  "inputs": [
+    {
+      "prevout": {
+        "hash": "a74c5a27a9d9ac13fd502dad0ed4971c40524fb7a7ca002fc30470997de9622c",
+        "index": 0
+      },
+      "script": "",
+      "witness": [
+        "",
+        "",
+        "3045022100804216dc296b0c0c0f1a1ed54751c60a349518f3d6a558927be0099ecdf10a3a022013cf0e5ddaf99b0d94c00e69c875f43f034ac7fb934b4abfb75f91e2a72b966d01",
+        "",
+        "",
+        "",
+        "53210242705b9570ec3c193ba67528ff38ee272f37b14ef33564e8d1b33812efb3ad812102bcb81bd3451ed30559473940b58e4272765f272dc17855e1be2b21a05945d23c2102d9f1f9f3214b64b31ef82c16376283e438d738406338776750f86424c2ec303c21036427eda147982560d066449070c457697cf22ff3bbe2d0ecd5ba070d52d9434a21039c6a894d0067c32997f2390c2ecffaa0376ac34c8c0a4ebfa6d738c28814456455ae"
+      ],
+      "sequence": 4294967295,
+      "coin": {
+        "version": 1,
+        "height": 1,
+        "value": 5000000000,
+        "script": "0020a17de28e0eaab5ec2570d25e054eebbd4b8ec26b973c89a753a2b92f516b1389",
+        "address": "bcrt1q59779rsw4267cfts6f0q2nhth49casntju7gnf6n52uj75ttzwys3zph8r",
+        "coinbase": true
+      }
+    }
+  ],
+  "outputs": [
+    {
+      "value": 123456789,
+      "script": "76a9145fb239be6a09c81b43591a7bbf5857424438052b88ac",
+      "address": "mpEwzcAKTsuGMM7hV49Mh3Ge7r933S6uYd"
+    },
+    {
+      "value": 4876538671,
+      "script": "0020b653b1dda1540084213f4cfa7e089e526f23b749da3dc8de09597ec085eefed6",
+      "address": "bcrt1qkefmrhdp2sqgggflfna8uzy72fhj8d6fmg7u3hsft9lvpp0wlmtq7uvkml"
+    }
+  ],
+  "locktime": 0,
+  "hex": "010000000001012c62e97d997004c32f00caa7b74f52401c97d40ead2d50fd13acd9a9275a4ca70000000000ffffffff0215cd5b07000000001976a9145fb239be6a09c81b43591a7bbf5857424438052b88ac2f13aa2201000000220020b653b1dda1540084213f4cfa7e089e526f23b749da3dc8de09597ec085eefed6070000483045022100804216dc296b0c0c0f1a1ed54751c60a349518f3d6a558927be0099ecdf10a3a022013cf0e5ddaf99b0d94c00e69c875f43f034ac7fb934b4abfb75f91e2a72b966d01000000ad53210242705b9570ec3c193ba67528ff38ee272f37b14ef33564e8d1b33812efb3ad812102bcb81bd3451ed30559473940b58e4272765f272dc17855e1be2b21a05945d23c2102d9f1f9f3214b64b31ef82c16376283e438d738406338776750f86424c2ec303c21036427eda147982560d066449070c457697cf22ff3bbe2d0ecd5ba070d52d9434a21039c6a894d0067c32997f2390c2ecffaa0376ac34c8c0a4ebfa6d738c28814456455ae00000000"
+}
+```
+
+Notice how the witness in the input of this transaction has several empty placeholders.
+When bcoin creates a multisig transaction, it places null bytes (`0x00`) where each
+signature will eventually go. The signatures must be placed in same order as the public
+keys in the redeem script, which is why Erica's signature is not the first on the stack.
```

Comment:
Link to bip 67 which explains the canonical public key ordering, otherwise each permutation of keys will result in different addresses and cosigners will be unable to locate their utxos.
https://github.com/bitcoin/bips/blob/master/bip-0067.mediawiki